tests/integration/security/util/reachability/context.go

```diff
@@ -171,69 +113,104 @@ func (rc *Context) Run(testCases []TestCase) {
 			// to work around this, we will temporarily make sure we are always sleeping at least 10s, even if istioctl wait is faster.
 			// This allows us to debug istioctl wait, while still ensuring tests are stable
 			sleep := time.Second*10 - time.Since(t0)
+			if !deployedWithinWait {
+				sleep = time.Second * 30
+			}
 			ctx.Logf("[%s] [%v] Wait for additional %v config propagate to endpoints...", testName, time.Now(), sleep)
 			time.Sleep(sleep)
 			ctx.Logf("[%s] [%v] Finish waiting. Continue testing.", testName, time.Now())
 			ctx.Cleanup(func() {
 				if err := retry.UntilSuccess(func() error {
-					return rc.ctx.Config().DeleteYAML(c.Namespace.Name(), policyYAML)
+					return ctx.Config().DeleteYAML(c.Namespace.Name(), policyYAML)
 				}); err != nil {
 					log.Errorf("failed to delete configuration: %v", err)
 				}
 			})
-
-			for _, src := range []echo.Instance{rc.A, rc.B, rc.Headless, rc.Naked, rc.HeadlessNaked} {
-				for _, dest := range []echo.Instance{rc.A, rc.B, rc.Headless, rc.Multiversion, rc.Naked, rc.VM, rc.HeadlessNaked} {
-					copts := &callOptions
-					// If test case specified service call options, use that instead.
-					if c.CallOpts != nil {
-						copts = &c.CallOpts
+			for _, clients := range []echo.Instances{apps.A, apps.B, apps.Headless, apps.Naked, apps.HeadlessNaked} {
```

Comment:
Seems like quite a lot of copy-paste from the `common/` package in pilot tests. We should probably prioritize consolidating this sooner rather than later.